configs/templates/generated_interfaces_for_adapters/list/ListProcessor.go

```diff
@@ -0,0 +1,14 @@
+package list
+
+import "istio.io/mixer/configs/templates/list"
+
+type Instance struct {
+  Blacklist       bool
+  CheckExpression interface{}
+}
+
+type ListProcessor interface {
+  ConfigureList(templateName string, types map[string]*foo_bar_mylistchecker.Type /*typeName to Type mapping*/) error
+  CheckList(templateName string, instances map[string]*Instance /*typeName to Instance (generated from Constructor) mapping*/) (bool, error)
```

Comment:
The two methods take different form on input. Which feels weird.
`ConfigureList` takes a proto object `foo_bar_mylistchecker.Type`, while `CheckList` takes a first class golang struct `Instance`.

Do we want to make the api signatures feel symmetrical ?

I also think the signature of these would change as we get closer to penning down the API signature for Disintermediate rpc and the rpcs that gRPC adapters would implement.

Thoughts ?